utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt

```diff
@@ -0,0 +1,33 @@
+package org.oppia.android.util.profile
+
+import android.content.Intent
+import org.oppia.android.app.model.CurrentUserProfile
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.util.extensions.getProtoExtra
+import org.oppia.android.util.extensions.putProtoExtra
+
+private const val PROFILE_ID_INTENT_DECORATOR = "PROFILE_ID_INTENT_DECORATOR"
+
+/** Decorator for wrapping a user's profile [ProfileId] within its intent. */
+object CurrentUserProfileIdIntentDecorator {
+  /**
+   * Packs the intent with a [ProfileId] proto object that sets it as the current
+   * screen.
+   *
+   * [extractCurrentUserProfileId] should be used for retrieving the screen name later.
+   */
+  fun Intent.decorateWithUserProfileId(profileId: ProfileId) {
+    putProtoExtra(
+      PROFILE_ID_INTENT_DECORATOR,
+      CurrentUserProfile.newBuilder().setProfileId(profileId).build()
```

Comment:
You could just store ProfileId directly, right? I don't think the extra proto is necessary.

Reply:
Thanks @BenHenning this makes sense, I have updated to use profileId directly instead. Have also added tests covering activities Intents utilizing ProfileId parameter.

Reply:
Fixed CI failures issues by adding the necessary missing bazel dependency and also updating tests to utilize the updated pattern.